eth/handler.go

```diff
@@ -400,13 +423,10 @@ func (pm *ProtocolManager) handle(p *peer) error {
 // peer. The remote connection is torn down upon returning any error.
 func (pm *ProtocolManager) handleMsg(p *peer) error {
 	// Read the next message from the remote peer, and ensure it's fully consumed
-	msg, err := p.rw.ReadMsg()
+	msg, err := p.ReadMsg()
```

Comment:
I see you created a new peer function `ReadMsg`.  Is there a specific reason for that?